fdbserver/DataDistribution.actor.cpp

```diff
@@ -2410,26 +2417,37 @@ ACTOR Future<Void> scheduleAuditOnRange(Reference<DataDistributor> self,
 					self->remainingBudgetForAuditTasks[auditType].set(SERVER_KNOBS->CONCURRENT_AUDIT_TASK_COUNT_MAX -
 					                                                  1);
 				}
-				audit->actors.add(doAuditOnStorageServer(self, audit, targetServer, req));
+				issueDoAuditCount++;
+				doAuditTasks.add(doAuditOnStorageServer(self, audit, targetServer, req));
 				// Proceed to the next range if getSourceServerInterfacesForRange is partially read
 				begin = rangeLocations[i].range.end;
 				wait(delay(0.1));
 			}
 		}
+
+		wait(doAuditTasks.getResult());
```

Comment:
If we `wait` on `doAuditTasks` and retry when there is an error, we could retry multiple sub-tasks due to one failure? Can we be more precise to retry only the failed range?

Reply:
Yes. Nice catch.

Reply:
Is it possible to call `scheduleAuditOnRange()` within `doAuditOnStorageServer()` when it fails, since that is the minimum range we know to retry. If we retry here with `range` passed to `scheduleAuditOnRange`, we could restart multiple `doAuditOnStorageServer`?